frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -234,13 +248,13 @@ function DashboardInner(props: DashboardProps) {
   const previousParameterValues = usePrevious(parameterValues);
 
   const currentTabDashcards = useMemo(() => {
-    if (!Array.isArray(dashboard?.dashcards)) {
+    if (!dashboard || !Array.isArray(dashboard?.dashcards)) {
       return [];
     }
     if (!selectedTabId) {
-      return dashboard.dashcards;
+      return dashboard?.dashcards;
     }
-    return dashboard.dashcards.filter(
+    return dashboard?.dashcards.filter(
       (dc: DashboardCard) => dc.dashboard_tab_id === selectedTabId,
     );
   }, [dashboard, selectedTabId]);
```

Comment:
Thes optional chaining additions shouldn't be needed as we check for `!dashboard` in the first `if`